### PR TITLE
misc cleanups from debugging something

### PR DESCRIPTION
**Pull request by BoxyUwU**
rename `instantiate_canonical_with_fresh_inference_vars` to `instantiate_canonical`  the substs for the canonical are not solely infer vars as that would be wildly wrong and it is rather confusing to see this method called and think that the entire canonicalization setup is completely broken when it is not :thumbsup:

also update region debug printing to be more like the custom impls for Ty/Const, right now regions in debug output are horribly verbose and make it incredibly hard to read but with this atleast boundvars and placeholders when debugging the new solver do not take up excessive amounts of space.

r? @lcnr

### CODE REVIEW

**Comment by matthiaskrgr** [comment]
this will also change some existing ice messages, right?
please don't forget to update the corresponding tickets to match the new scheme once this lands, otherwise people may open duplicates by accident when they can't find the old tickets when searching for the update ice messages. :slightly_smiling_face:   thanks!

**Review by lcnr — COMMENTED** [review]
maybe just `fn instantiate_canonical` and change the first line of its doc comment to mention both placeholders and infer vars?

The function name as is feels a bit too long to me

**Comment by BoxyUwU** [comment]
@matthiaskrgr yes this could change ICE output. I'm not sure how widespread tbh, I can't remember if we ever made it so that ICE output that debug prints Ty's will actually debug print them or not. Regardless I have no idea how we would find all the ICEs out of the thousands of issues that contain old region debug output and update it. Not going to block this PR on that as that is a huge burden for simply changing printing used for debugging rustc.

File: compiler/rustc_middle/src/ty/region.rs [hunk 1]
```diff
@@ -336,7 +336,9 @@ pub struct EarlyParamRegion {
 
 impl std::fmt::Debug for EarlyParamRegion {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        write!(f, "{:?}, {}, {}", self.def_id, self.index, self.name)
+        // self.def_id goes first because of `erased-regions-in-hidden-ty.rs` being impossible to write
```
Comment (lcnr) [inline compiler/rustc_middle/src/ty/region.rs:339]: make this a FIXME, this is not a good reason. I would expect the preferred repr to be name, index, `def_id`

File: compiler/rustc_middle/src/ty/region.rs [hunk 2]
```diff
@@ -381,13 +383,25 @@ pub enum BoundRegionKind {
...
 pub struct BoundRegion {
     pub var: BoundVar,
     pub kind: BoundRegionKind,
 }
 
+impl core::fmt::Debug for BoundRegion {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self.kind {
+            BoundRegionKind::BrAnon => write!(f, "{:?}", self.var),
+            BoundRegionKind::BrEnv => write!(f, "{:?}_Env", self.var),
```
Comment (lcnr) [inline compiler/rustc_middle/src/ty/region.rs:397]:
```suggestion
            BoundRegionKind::BrEnv => write!(f, "{:?}.Env", self.var),
```
personal preference, feel free to ignore

File: compiler/rustc_middle/src/ty/region.rs [hunk 3]
```diff
@@ -336,7 +336,9 @@ pub struct EarlyParamRegion {
 
 impl std::fmt::Debug for EarlyParamRegion {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        write!(f, "{:?}, {}, {}", self.def_id, self.index, self.name)
+        // self.def_id goes first because of `erased-regions-in-hidden-ty.rs` being impossible to write
+        // error annotations for otherwise. :)
+        write!(f, "({:?})_{}/#{}", self.def_id, self.name, self.index)
```
Comment (lcnr) [inline compiler/rustc_middle/src/ty/region.rs:341]:
```suggestion
        write!(f, "{:?}_{}/#{}", self.def_id, self.name, self.index)
```
isnÄ't it `DefId(whatever)` anyways?

File: compiler/rustc_type_ir/src/region_kind.rs [hunk 4]
```diff
@@ -223,23 +223,24 @@ impl<I: Interner> DebugWithInfcx<I> for RegionKind<I> {
...
-            ReStatic => f.write_str("ReStatic"),
+            ReStatic => f.write_str("'static"),
 
             ReVar(vid) => write!(f, "{:?}", &this.wrap(vid)),
 
-            RePlaceholder(placeholder) => write!(f, "RePlaceholder({placeholder:?})"),
+            RePlaceholder(placeholder) => write!(f, "{placeholder:?}"),
 
-            ReErased => f.write_str("ReErased"),
+            ReErased => f.write_str("'erased"),
```
[the remark this answers is not in the document]
Comment (BoxyUwU) [reply compiler/rustc_type_ir/src/region_kind.rs:241]: I think actual regions named 'erased would be formatted like `'erased/#2` or wahtever so its not technically ambiguous but I'll change this anyway

File: compiler/rustc_middle/src/ty/region.rs (hunk 3 above)
Comment (BoxyUwU) [reply compiler/rustc_middle/src/ty/region.rs:341]: yeah ur right

File: compiler/rustc_middle/src/ty/region.rs (hunk 1 above)
Comment (BoxyUwU) [reply compiler/rustc_middle/src/ty/region.rs:339]: agreed idk why I didn't make this a FIXME

**Comment by lcnr** [comment]
r=me after blessing mir opt tests and CI is green

**Pull request merged**